modules/nix/default.nix

```
@@ -831,18 +831,28 @@ in
       rm --force $out/bin/nix-channel
     '';
 
-    nix.nrBuildUsers = mkDefault (max 32 (if cfg.settings.max-jobs == "auto" then 0 else cfg.settings.max-jobs));
+    nix.nrBuildUsers = mkDefault (
+      if !configureBuildUsers then
+        0
+      else
+        max 32 (if cfg.settings.max-jobs == "auto" then 0 else cfg.settings.max-jobs)
+    );
 
     users.users = mkIf configureBuildUsers nixbldUsers;
 
     # Not in NixOS module
-    users.groups.nixbld = mkIf configureBuildUsers {
+    users.groups.nixbld = {
       description = "Nix build group for nix-daemon";
       gid = config.ids.gids.nixbld;
       members = attrNames nixbldUsers;
     };
     users.knownUsers =
-      let nixbldUserNames = attrNames nixbldUsers;
+      let
+        # This creates 128 "known" build users -- in other words, any user
+        # between `_nixbld1` and `_nixbld128` is considered "managed" by
+        # nix-darwin, and will be created/deleted as `nrBuildUsers` increases
+        # and decreases (or is set to 0 in case of `auto-allocate-uids`.
+        nixbldUserNames = map (x: "_nixbld${toString x}") (range 1 128);
```

```suggestion
        nixbldUserNames = map (x: "_nixbld${toString x}") (range 1 maxBuildUsers);
```
and use that for a `lib.types.ints.between 1 maxBuildUsers` type in `nrBuildUsers`.